src/compas_rhino/geometry/brep/brep.py

```diff
@@ -0,0 +1,82 @@
+from compas.geometry import Geometry
+from compas_rhino.conversions import box_to_rhino
+
+from Rhino.Geometry import Brep
+
+from .face import RhinoBRepFace
+from .edge import RhinoBRepEdge
+from .vertex import RhinoBRepVertex
+from .loop import RhinoBRepLoop
+
+
+class RhinoBRep(Geometry):
+    def __init__(self):
+        super(RhinoBRep, self).__init__()
+        self.rhino_brep = None
+
+    @property
+    def data(self):
+        faces = []
+        for face in self.faces:
+            faces.append(face.data)
+        return {"faces": faces}
+
+    @data.setter
+    def data(self, data):
+        faces = []
+        for facedata in data["faces"]:
+            face = RhinoBRepFace.from_data(facedata)
+            faces.append(face)
+        self.rhino_brep = BRep.from_faces(faces).rhino_brep
+
+    @classmethod
+    def from_faces(cls, faces):
+        brep = Brep()
+
+    @property
+    def points(self):
+        if self.rhino_brep:
+            return [RhinoBRepVertex(v) for v in self.rhino_brep.Vertices]
+
+    @property
+    def edges(self):
+        if self.rhino_brep:
+            return [RhinoBRepEdge(e) for e in self.rhino_brep.Edges]
+
+    @property
+    def loops(self):
+        if self.rhino_brep:
+            return [RhinoBRepLoop(l) for l in self.rhino_brep.Loops]
+
+    @property
+    def faces(self):
+        if self.rhino_brep:
+            return [RhinoBRepFace(f) for f in self.rhino_brep.Faces]
+
+    @property
+    def frame(self):
+        """TODO: what to do here? Brep has no Frame/plane information"""
```

Comment:
in Rhino?
i guess that means the frame is `Frame.worldXY()`

Reply:
i think it will be difficult to use a local frame for the brep objects, because it means not only the brep topology elements but also the underlying (nurbs) geometry will have to be expressed wrt to this local frame.

for the shapes, we have to make a distinction between parametric shapes (box, sphere, ...) and the non-parametric ones (i guess this is only polyhedron, which is a special case also in other regards). all the parametric ones should have a local frame. transformations are easy because you just transform the frame, and world coordinates can be calculated based on the parameters on the fly.

data structures (and polyhedron) are defined explicitly wrt to the global coordinate system.

Reply:
would do you mean by "keeping track of transformations"? transformations defining subsequent movements of a geometry in space? currently we differentiate between the geometry of an object and the viewobject (a bit like in Rhino and Blender). the viewobject can store a transformation stack that can keep track of such transformations...

Reply:
well i would argue you calculate the actual frame to frame transformation for each of those, and store the matrices in a transformation stack. but, again, not sure if this is something that should be added to the geometry itself...

Reply:
about the (non-)parametric shapes. this goes a bit in the direction of the discussion about adding a reference frame to all shapes.

i think in COMPAS 2.0 (or in the LTS version of 1.0, but i don't know if we manage to do that without breaking changes) we should differentiate between the following objects.

* Primitives: point, vector, line, plane, frame, polyline, polygon (defined by point locations), polyhedron (defined by polygons)
* Parametric shapes: circle, ellipse, regular polygon (defined by a radius and the number of sides), regular polyhedron (more or less the platonic solids), box, sphere, torus, cylinder, cone, ellipsoid
* Nurbs: curve, surface
* Boundary representations
* Function representations (not sure about this one)
* Transformations

the parametric shapes are implicitly defined wrt a local frame. all other objects are explicitly defined wrt the global coordinate system.

anyway, i know this is not part of this PR, but since we were talking about it...

Reply:
sounds like a good categorization, we should eventually (ie. outside this PR) add `ParametricShape` and `Nurb` as a base classes. I believe all others already exist as classes. re. function representation, is that `compas_vol` sort of thing?

Reply:
yes indeed. perhaps it is a bit too specific, but some base infrastructure could easily be added in core...